device/simulation/tt_simulation_host.cpp

```diff
@@ -0,0 +1,72 @@
+// SPDX-FileCopyrightText: (c) 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include <typeinfo>
+#include <sstream>
+#include <iomanip>
+#include <filesystem>
+#include <cassert>
+
+#include "common/logger.hpp"
+#include "common/assert.hpp"
+#include "tt_simulation_host.hpp"
+
+tt_SimulationHost::tt_SimulationHost() {
+    // Open socket and create dialer
+    nng_pair1_open(&host_socket);
+    std::string nng_socket_addr_str;
+    const char *ipc_socket_name = std::getenv("NNG_SOCKET_NAME");
+    if (!ipc_socket_name) {
+        nng_socket_addr_str = std::string(NNG_SOCKET_PREFIX) + std::string("nng_ipc_socket");
```

Comment:
Any way you can make this unique automatically? Unique by user for sure, but unique per test run would be ideal.

I see a potential issue if multiple users run same test, or same user runs multiple tests, it could end up with different host code targeting same simulator instance. 
As far as I can see, you assume here that user will always setup different NNG_SOCKET_NAME, and in multi user env, I fully expect that someone will setup that the same.

Reply:
Will now create unique socket addr that will be in the form of: 
`ipc:///tmp/<username>_<date/time>_<NNG_SOCKET_NAME>` 

where username is just `whoami` or `echo $USER` value on the machine.

eg:
`ipc:///tmp/vtang_08-02-16:00:34_deviceinit`

this should ensure nothing targets the same simulator instance as the date/time will be at seconds granularity

If `NNG_SOCKET_NAME` is not set, the default is `ipc:///tmp/<username>_<date/time>_nng_ipc`